src/components/PaymentInfoLine.vue

```diff
@@ -82,12 +167,95 @@ class PaymentInfoLine extends Vue {
         default: 'normal',
     })
     public theme!: string;
+    private referenceRate: number | null = null;
+    private referenceRateUpdateTimeout: number = -1;
+    private lastTooltipToggle: number = -1;
+
+    private created() {
+        this.updateReferenceRate();
+    }
+
+    private destroyed() {
+        window.clearTimeout(this.referenceRateUpdateTimeout);
+    }
 
     public async setTime(time: number) {
         await this.$nextTick(); // let vue update in case the timer was just added
         if (!this.$refs.timer) return;
         (this.$refs.timer as Timer).synchronize(time);
     }
+
+    private get originDomain() {
+        return this.origin.split('://')[1];
+    }
+
+    private get effectiveRate() {
+        if (!this.fiatAmount) return null;
+        // Note: precision loss should be acceptable here
+        return this.fiatAmount.amount / (Number(this.cryptoAmount.amount) / (10 ** this.cryptoAmount.decimals));
+    }
+
+    private get formattedMerchantFee() {
+        if (this.merchantFee === null || this.merchantFee === undefined) return null;
+        // Note: precision loss should be acceptable here
+        const merchantFeePercent = Number(this.merchantFee) / Number(this.cryptoAmount.amount) * 100;
```

Comment:
This calculation is not correct.

Example: `cryptoAmount = 2000, merchantFee = 500`. This calculation computes a 25% merchant rate.

But actually, the merchantFee is already _included_ in the cryptoAmount, so you have to subtract it first. This means the actual item price is `2000 - 500 = 1500`. So the actual merchant rate would be `500 / 1500 = 33%`.

Note that this calculation is not needed anymore if the merchant rate prop is accepted in percent.